api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2TokenRefreshExecutor.java

```diff
@@ -28,8 +28,14 @@ final class OAuth2TokenRefreshExecutor extends ScheduledThreadPoolExecutor
 
   private static final Logger LOGGER = LoggerFactory.getLogger(OAuth2TokenRefreshExecutor.class);
 
+  // Need multiple threads:
+  // - One to run org.projectnessie.client.auth.oauth2.OAuth2Client.renewTokens
+  // - One to run org.projectnessie.client.auth.oauth2.DeviceCodeFlow.pollForNewTokens
+  // - (a spare one - just in case) ;)
+  private static final int CORE_POOL_SIZE = 3;
```

Comment:
Heh, this resonates with my comment above. But I am not convinced we need that many threads, 1 is probably enough because the core size is 1, but the max size is unbounded. And as I said before, the thread that runs the device code flow is a short-lived one.

Reply:
Also, not entirely related to this change, but I wonder if we need to keep `TokenRefreshThread`. The only 2 things it does is: give a name to the thread, and some logging before/after execution. 

First off, the thread name could be managed by the factory:

```java
Thread thread = new Thread(r, "nessie-client-oauth2-token-refresh");
```

Secondly, the thread name is fixed, which is not great.

And lastly, the logging is questionable imho (I think I did this for debugging purposes when developing this.)